src/backend/joanie/core/models/contracts.py

```diff
@@ -226,6 +226,7 @@ def tag_submission_for_signature(self, reference, checksum, context):
         self.definition_checksum = checksum
         self.signature_backend_reference = reference
         self.save()
+        self.order.flow.update()
 
     def reset_submission_for_signature(self):
         """
```

Comment:
I think we should also try to update the flow on reset_submission to go back to `to_sign` state